lib/livebook/hubs/team_client.ex

```diff
@@ -96,9 +108,40 @@ defmodule Livebook.Hubs.TeamClient do
     {:noreply, %{state | connected?: false}}
   end
 
+  def handle_info({:event, topic, data}, state) do
+    Logger.debug("Received event #{topic} with data: #{inspect(data)}")
+
+    {:noreply, handle_event(topic, data, state)}
+  end
+
   # Private
 
   defp registry_name(id) do
     {:via, Registry, {@registry, id}}
   end
+
+  defp put_secret(state, secret) do
+    %{state | secrets: [secret | state.secrets]}
+  end
+
+  defp build_secret(state, %{name: name, value: value}) do
+    {secret_key, sign_secret} =
+      Livebook.Stamping.derive_keys(state.hub.teams_key, "notebook secret")
+
+    {:ok, decrypted_value} = Plug.Crypto.MessageEncryptor.decrypt(value, secret_key, sign_secret)
```

Comment:
@jonatanklosko IIRC you should always sign when you encrypt because encryption by itself does not guarantee it was not tampered with (or something like that).

Reply:
Ah right, we still want want authenticated encryption, we just don't need the additional data, sounds good!